frontends/election-manager/src/hooks/use_election_manager_store.ts

```diff
@@ -0,0 +1,697 @@
+      await logger.log(LogEventId.LoadFromStorage, currentUserRoleRef.current, {
+        message: 'Failed to parse printed ballots from storage',
+        disposition: 'failure',
+        storageKey: printedBallotsStorageKey,
+        error: parseResult.err().message,
+      });
+      return;
+    }
+
+    const printedBallots = parseResult.ok();
+
+    await logger.log(LogEventId.LoadFromStorage, currentUserRoleRef.current, {
+      message: 'Printed ballots automatically loaded from storage',
+      disposition: 'success',
+      storageKey: printedBallotsStorageKey,
+      totalCount: printedBallots.reduce(
+        (count, printedBallot) => count + printedBallot.numCopies,
+        0
+      ),
+    });
+
+    return printedBallots;
+  }, [logger, storage]);
+
+  const getPrintedBallotsQuery = useQuery<PrintedBallot[]>(
+    ['printedBallots'],
+    async () => {
+      return (await loadPrintedBallots()) ?? [];
+    }
+  );
+  const printedBallots = getPrintedBallotsQuery.data;
+
+  const loadElectionDefinition = useCallback(async (): Promise<
+    ElectionDefinition | undefined
+  > => {
+    const loadedElectionDefinition = (await storage.get(
+      electionDefinitionStorageKey
+    )) as ElectionDefinition | undefined;
+
+    if (loadedElectionDefinition) {
+      const parseResult = safeParseElectionDefinition(
+        loadedElectionDefinition.electionData
+      );
+
+      if (parseResult.isErr()) {
+        await logger.log(LogEventId.LoadFromStorage, 'system', {
+          message: 'Error parsing election definition from storage',
+          disposition: 'failure',
+          storageKey: electionDefinitionStorageKey,
+          error: parseResult.err().message,
+        });
+        return;
+      }
+
+      const parsedElectionDefinition = parseResult.ok();
+
+      if (
+        parsedElectionDefinition.electionHash !==
+        loadedElectionDefinition.electionHash
+      ) {
+        await logger.log(LogEventId.LoadFromStorage, 'system', {
+          message: 'Election definition hash mismatch',
+          disposition: 'failure',
+          storageKey: electionDefinitionStorageKey,
+          expectedElectionHash: loadedElectionDefinition.electionHash,
+          actualElectionHash: parsedElectionDefinition.electionHash,
+        });
+        return;
+      }
+
+      await logger.log(LogEventId.LoadFromStorage, currentUserRoleRef.current, {
+        message: 'Election definition automatically loaded from storage',
+        disposition: 'success',
+        storageKey: electionDefinitionStorageKey,
+        electionHash: parsedElectionDefinition.electionHash,
+      });
+
+      return parsedElectionDefinition;
+    }
+  }, [logger, storage]);
+
+  const getElectionDefinitionQuery = useQuery<ElectionDefinition | null>(
+    [electionDefinitionStorageKey],
+    async () => {
+      return (await loadElectionDefinition()) ?? null;
+    }
+  );
+  const electionDefinition = getElectionDefinitionQuery.data ?? undefined;
```

Comment:
Why the pattern of having the `useQuery` return `TData | null` and then replacing `null` with `undefined` here?

Reply:
Ah, I guess it's slightly incompatible with our conventions around `null` and `undefined`. An explanatory comment might be helpful for future reference

Reply:
Will add a comment.

Reply:
Comment added.